tests/functional/test_install_config.py

```diff
@@ -416,22 +417,32 @@ def test_prompt_for_keyring_if_needed(
             "keyring",
         )
 
-        environ["PATH"] = str(keyring_script.bin_path) + os.pathsep + environ["PATH"]
-
-    virtualenv = virtualenv_factory(workspace.joinpath("venv"))
-    script = script_factory(workspace.joinpath("venv"), virtualenv, environ=environ)
-
-    if (
-        keyring_provider not in [None, "auto"]
```

Comment:
I found the pre-existing code hard to grok. This line in particular really confounds me: why does `keyring_provider` influence how we set up the test environment? I'd expect that to be only dependent upon `keyring_provider_implementation` (which is the change I've made in this PR).

Reply:
> After your refactoring keyring won't be importable when keyring_provider in [None, "auto"] and keyring_provider_implementation == "disabled".

Yes, this was intentional. Perhaps I'm misunderstanding these variable names? I read them as:

- `keyring_provider_implementation`: this is the way i'm going to set up my environment for this test (is keyring installed in this venv, is is discoverable via PATH in a different venv, or is it not installed at all)
- `keyring_provider`: this is the keyring provider i'm going to request when i invoke pip

So, it's intentional that if `keyring_provider_implementation == "disabled"`, then yeah, keyring won't be imprtable.

But again, it's very possible I'm misunderstanding the intent of these tests. Guidance appreciated.

> Maybe it is an idea to ask for verbose output and check for the Keyring provider set: lines to make that more explicit?

Yeah, I personally not in love with this elaborate usage of parameterize + xfail. Tests can fail for all sorts of reasons other than the expected reason! I think it would be better to assert on the expected failures. (IMO, that would be better to do in a separate PR)